cni-plugin/integration/flannel/flannel_test.go

```diff
@@ -0,0 +1,93 @@
+package flannel
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+)
+
+const (
+	ConfigDirectory = "/var/lib/rancher/k3s/agent/etc/cni/net.d"
+	FlannelConflist = "10-flannel.conflist"
+)
+
+func ls(dir string, t *testing.T) []string {
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		t.Fatalf("failed to list files: %v", err)
+	}
+	fileNames := make([]string, len(files))
+	for i, f := range files {
+		fileNames[i] = f.Name()
+	}
+	return fileNames
```

Comment:
Ok...unpopular opinion time, if we do something unorthodox here and store the names as keys in a map we can avoid `contains` and we can simply do one-liners throughout the code 

```go
if _, ok := mymap[FlannelConflist]; !ok {
  t.Fatalf("%s not found", FlannelConflist)
}
``` 

We can limit the amount of space we use by storing `struct{}` ([which has 0 bytes in sz!](https://dave.cheney.net/2014/03/25/the-empty-struct)):

```go
fileNames := make(map[string]struct{}, len(files))
```

I can understand if you do not like this approach though and will not take it personal 😂

Reply:
Using a map with empty keys here is a good idea, better than my 1970's approach for sure. I'll change it and also change these functions to return err instead of passing in a `*testing.T`